packages/next-auth/src/providers/battlenet.ts

```diff
@@ -0,0 +1,32 @@
+import type { OAuthConfig } from "."
+import OAuthProvider from './auth0'
+export interface BattleNetProfile {
+  sub: string
+  battle_tag: string
+}
+
+export default function BattleNet(
+  options: { clientId: string, clientSecret: string, region: 'US' | 'EU' | 'KR' | 'TW' | 'CN' }
+): OAuthConfig<BattleNetProfile> {
+
+  const wellKnown = options.region === 'CN' 
+    ? 'https://www.battlenet.com.cn/oauth/.well-known/openid-configuration' 
+    : `https://us.battle.net/oauth/.well-known/openid-configuration` 
```

Comment:
```suggestion
  const { region = "CN" } = options
  const domain =
    region === "CN"
      ? "https://www.battlenet.com.cn"
      : `https://${region}.battle.net`
```

Reply:
Considering region isn't optional, why would we default it to CN here?

Reply:
Sorry, I assumed it was the case before, but it isn't true. https://github.com/nextauthjs/next-auth/blob/3d0c68d9e00c707f254d17a511c54af4ebb9eeb0/packages/next-auth/src/providers/battlenet.js#L3-L7